### ISSUE TITLE
Feature request: Twitch and Discord bungieMembershipTypes

### ISSUE THREAD

**Issue by floatingatoll** [Bungie-net/api#520]
Please consider offering Bungie.net API (and website and auth) integration with Twitch and Discord. It would make it possible for clans to link up their members on these services without having to create third-party databases of these links, and ensure that players rather than third-party services are in control of these links.

**Comment by Tetron-bng** [comment]
The partnership endpoint can reveal Twitch accounts linked to a user's Bungie.net membership.  I don't remember if we made that public or not.

https://bungie-net.github.io/multi/operation_get_User-GetPartnerships.html#operation_get_User-GetPartnerships

**Comment by nine13tech** [comment]
This is public. I use it to show which of my clan members have stream accounts.
@floatingatoll I use the API data as an overwritten at each pull set of fields in my DB to keep track - the endpoint gives me the platform (twitch) and the screen name
From what I have seen there is no discord linking.

**Comment by Tetron-bng** [comment]
Yeah, no support for Discord.  But this might be a logical place to put it.  This would avoid adding non-Destiny platform sign-in methods, which is a much bigger can of worms.  If this works for you with Twitch it could work for other platforms with OAuth support, such as Discord. Look at me signing Alex up for work. (ducking back out)

**Comment by vthornheart-bng** [comment]
Ha!  @Tetron-bng does this mean you're coming back to work on it? :)

I'd love to add Discord linkage!  But we'd need a good user-facing feature to justify it to design before we could get it prioritized over other features on the backend.  Ultimately because we're pretty short staffed on the server side, we end up prioritizing work for upcoming Companion features or game integration needs over features that we don't have immediate plans for in our app.  But that being said, I would like to see it come around: I just don't think it will likely make its way out of the backlog with the current workload.  I'm going to put this on the distant future wishlist!

**Comment by ckhicks** [comment]
> Ultimately because we're pretty short staffed on the server side, we end up prioritizing work for upcoming Companion features or game integration needs over features that we don't have immediate plans for in our app.

This is both insightful and very helpful information! We owe you guys coffee, big time.

**Comment by floatingatoll** [comment]
https://careers.bungie.com/en-US/Careers/IT-Operations is helpful for that question!

**Comment by Tetron-bng** [comment]
More like: https://careers.bungie.com/en-US/Careers/Engineering . But I don't see my old job listed.

**Comment by migit128** [comment]
Yea, already checked both those pages out. I didnt see anything that screamed "API"

**Comment by floatingatoll** [comment]
Check back occasionally — my employees over time can often take several weeks after someone departs to put up a posting — and worst case scenario, you’re given an offer for a non-API role instead and have to decide whether to accept or reject it!

**Comment by migit128** [comment]
It was more out of curiosity than anything.  If I was to take a job like this I'd spend all my time at work doing the stuff they want, then spending my weekends doing the stuff I want done.  Probably not worth uprooting my life to waste away working on the API...  It'd be so nice to have actual clan management features built into bungie.net though and not have to build your own.

This isnt off topic at all! :P
